packages/webview_flutter/lib/request.dart

```diff
@@ -0,0 +1,37 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Represents a request made by the [WebView] widget.
+///
+/// See also:
+/// * [WebView.initialRequest] for how this gets used.
+class Request {
+  /// Constructs a [Request] object with the given [url]
+  /// and [headers].
+  const Request({
+    this.url,
+    this.headers,
+  });
+
+  /// The URL to load.
+  final String url;
+
+  /// The headers  to be passed in when making the request.
+  final Map<String, String> headers;
+
+  Map<String, dynamic> toMap() {
+    final Map<String, dynamic> optionsMap = <String, dynamic>{};
+
+    void addIfNonNull(String fieldName, dynamic value) {
```

Comment:
Could you include all the fields even if they're null and check for null on the native side? Or assert that `headers != null` if you implement my suggestion above about having a default value.